content/connectors/developer-guide/overview.md

```diff
@@ -2,3 +2,99 @@
 title: Overview
 weight: 10
 ---
+
+Writing a new connector is meant to be easy. You may write your connector in
+any language which has a [fluvio client library](/api). You should simply
+follow a few conventions and will gladly add and maintain a connector in our
+official catalog.
+
+## Adding a new connector
+
+A new connector should work as a stand alone program in development but when
+merged into the [`fluvio-connectors`
+repo](https://github.com/infinyon/fluvio-connectors/), this will be built into
+a docker image. It is recommended to write your connector in Rust and add a new
+[package in the
+workspace](https://doc.rust-lang.org/book/ch14-03-cargo-workspaces.html#creating-the-second-package-in-the-workspace)
+in the
+[`rust-connectors/sources`](https://github.com/infinyon/fluvio-connectors/tree/main/rust-connectors/sources)
+or `rust-connectors/sinks` directories.
+
+### `metadata` subcommand
+
+A connector should have a `metadata` command which prints a [json
+schema](https://json-schema.org/) of the commandline arguments. This is the
+command we use to build our connectors library and validate arguments passed to
+a connector.
+
+This `metadata` subcommand should print to stdout something of the following:
+```json
+{
+    "name": "test-connector",
+    "direction": "Source",
+    "schema": {
```

Comment:
I would recommend showing the command that printed this output, e.g.

```
$ test-connector metadata
{ ... }
```

that way people have more context on the fact that it's a subcommand to their connector executable